addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRule.java

```diff
@@ -304,13 +350,17 @@ private void localFileReflectionAttack(HttpMessage msg) {
      * @param msg new HttpMessage with the same request as the base. This is used to build the
      *     attack payload.
      */
-    private void localFileInclusionAttack(HttpMessage msg) {
+    private void localFileInclusionAttack(HttpMessage msg, String param) {
         String payload = null;
         try {
             for (int idx = 0; idx < LOCAL_FILE_TARGETS.length; idx++) {
                 String localFile = LOCAL_FILE_TARGETS[idx];
                 payload = MessageFormat.format(ATTACK_HEADER + ATTACK_BODY, localFile);
-                msg.setRequestBody(payload);
+                if (param != null) {
+                    this.setParameter(msg, param, payload);
```

Comment:
This is not creating the expected request. It is appending text to the request in the wrong location after the first cycle. I believe this is happening because the msg object is changed on each cycle and the `this.setParameter(` is using the `NameValuePair` made to the original request. Since the request is not the original the indexes used may not match. If you look to the request the scanner is creating you can see they have a strange format. One solution would be to get a new copy of the original request on each of the cycles. `HttpMessage msg = getNewMsg();`

Reply:
Thanks for testing this. Yes using getNewMsg() is the right choice (as far as I see)

Reply:
Right, messages shouldn't be reused.